### PR TITLE
feat: [SSCA-554] slsa provenance at stage level CI

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

form validations and feature behind FF SSCA-553
SSCA-555 slsa provenance step details under execution artifact details

#### Screenshots

image

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-ci/components/PipelineStudio/BuildStageSpecifications/BuildStageSpecifications.tsx [hunk 1]
```diff
@@ -102,6 +104,15 @@ export default function BuildStageSpecifications({ children }: React.PropsWithCh
...
     variables: NGVariable[]
+    slsa_provenance?: {
+      enabled: boolean,
+      attestation:{
+        type: 'cosign'
+        spec:{
+          password: string
+          private_key: string
+        }}
+      }
```
Comment (udayvunnam) [inline src/modules/75-ci/components/PipelineStudio/BuildStageSpecifications/BuildStageSpecifications.tsx:115]: todo, update this with BE generated type.

File: src/modules/75-ci/components/PipelineStudio/BuildStageSpecifications/BuildStageSpecifications.tsx [hunk 2]
```diff
@@ -140,9 +151,10 @@ export default function BuildStageSpecifications({ children }: React.PropsWithCh
       tags,
       cloneCodebase,
       sharedPaths: sharedPaths as any,
-      cacheIntelligenceEnabled: cacheIntelligenceEnabled,
+      cacheIntelligenceEnabled,
       cacheIntelligencePaths,
       cacheIntelligenceKey,
+      slsa_provenance: (spec as any)?.slsa_provenance,
```
Comment (udayvunnam) [inline src/modules/75-ci/components/PipelineStudio/BuildStageSpecifications/BuildStageSpecifications.tsx:157]: TODO

**Review by ishantmehtaharness — DISMISSED** [review]
looks good

**Review by ishantmehtaharness — APPROVED** [review]
LGTM

**Review by shaurya-harness — APPROVED** [review]

**Pull request merged**